### PR TITLE
add julia testing frameworks

### PR DESCRIPTION
**Pull request by BeastyBlacksmith**
### Summary

Adds testing frameworks for the julia programming language.

Also fixes formatting of codeblocks I came accross

### Acknowledging contributors

- [x] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.

### CODE REVIEW

**Comment by bouweandela** [comment]
I'm not much of a Julia expert, but my colleague @abelsiqueira is. Could you please have a look at this @abelsiqueira?

File: book/website/reproducible-research/testing/testing-exceptions.md [hunk 1]
```diff
@@ -182,4 +182,5 @@ Unit test frameworks for other languages also often provide similar functions:
 - JUnit for Java: org.junit.Assert.assertEquals(double expected, double actual, double delta)
 - testthat for R:
   - expect_equal(actual, expected, tolerance=DELTA) - absolute error within DELTA
-  - expect_equal(actual, expected, scale=expected, tolerance=DELTA) - relative error within DE L T A
+  - expect_equal(actual, expected, scale=expected, tolerance=DELTA) - relative error within DELTA
+- julia: `isapprox` or `≈`
```
Comment (abelsiqueira) [inline book/website/reproducible-research/testing/testing-exceptions.md:186]: I suggest adding them as list and showing the parameters, like
- `val1 ≈ val2`
- `isapprox(val1, val2, atol=abs_delta, rtol=rel_delta)`
- Test.jl with `≈`: `@test val1 ≈ val2 atol=abs_delta rtol=rel_delta`

**Review by abelsiqueira — APPROVED** [review]
Thanks for updating it @BeastyBlacksmith, looks good to me.

**Comment by abelsiqueira** [comment]
@bouweandela I have reviewed and approved it, but I don't think I have write access so you might need to approve it as well.

**Comment by bouweandela** [comment]
Thanks for reviewing @abelsiqueira! I do not have the right (or the ability to judge) if this is ready to be merged either. Maybe @malvikasharan could have a look?

**Comment by arronlacey** [comment]
Hi @BeastyBlacksmith, @abelsiqueira , @bouweandela - I agree that this checks out.

**Review by arronlacey — APPROVED** [review]

**Comment by malvikasharan** [comment]
Merging upon request. Some checks seem to fail but because of some code snippet that uses Hash symbol - #, that netlify takes it as Markdown header.

**Comment by malvikasharan** [comment]
@all-contributors please add @BeastyBlacksmith for idea and content.

**Comment by malvikasharan** [comment]
https://github.com/all-contributors please add @bouweandela for review

**Comment by malvikasharan** [comment]
@all-contributors please add @bouweandela for review

**Comment by malvikasharan** [comment]
@all-contributors please add @abelsiqueira  for review

**Comment by malvikasharan** [comment]
@all-contributors please add @arronlacey  for review

**Pull request merged**